contracts/WildcardSteward_v3_matic.sol

```diff
@@ -970,15 +976,15 @@ contract WildcardSteward_v3_matic is Initializable, BasicMetaTransaction {
         require(now >= tokenAuctionBeginTimestamp[tokenId], "not on auction");
         uint256 auctionTokenPrice = _auctionPrice(tokenId);
 
-        // uint256 remainingValueForDeposit = msg.value.sub(auctionTokenPrice);
-
         _distributeAuctionProceeds(tokenId);
 
         state[tokenId] = StewardState.Owned;
 
         serviceProviderPercentages[tokenId] = serviceProviderPercentage;
-        receiveErc20(depositAmount.add(auctionTokenPrice), msgSender());
-        deposit[msgSender()] = deposit[msgSender()].add(depositAmount);
+        receiveErc20(value, msgSender());
+        deposit[msgSender()] = deposit[msgSender()].add(value).sub(
+            auctionTokenPrice
+        );
```

Comment:
This is important

Reply:
Yup this seems more simple.

We might need to add another require that value > Auction price.

Same goes for the other Buy function, value > currentPrice.

Otherwise I guess the user is using part of their deposit to actually fund the purchase, which is maybe fine? But it will revert if their deposit cannot cover the shortfall

Reply:
Yeah, that is fine, it is also using safe maths, so it will definitely revert, although not with the nicest error message :/

Well I think for now it is fine :)

Reply:
Cool, yeah I thought you might have designed it like that intentionally. Think its good. 

Note if a user is using part of their deposit to fund their purchase, the front end might restrict them from this? As we normally check the users balance, and we don't display the buy button if their balance is less that the wildcard price.